php/utils.php

```diff
@@ -323,15 +323,36 @@ function loadFile($path) {
  */
 function parseUrl($url) {
   $url_parts = parse_url($url);
-
   if (!isset($url_parts['scheme'])) {
     $url_parts = parse_url('http://' . $url);
   }
-
   return $url_parts;
 }
 
 /**
+ * Opens a URL in a browser window
+ *
+ * @param string $url URL to open in a new browser window
+ * @return int
+ */
+function openInBrowser($url) {
+  switch (php_uname('s')) {
+    case 'Linux':
+      $cmd = 'xdg-open';
+        break;
+    case 'Darwin':
+      $cmd = 'open';
+        break;
+    case 'Windows NT':
+      $cmd = 'start';
+        break;
+  }
+  $command = sprintf('%s %s', $cmd, $url);
```

Comment:
I think the url needs to be shell escaped rather than url encoded:

http://php.net/manual/en/function.escapeshellarg.php